File: src/main/java/krasa/editorGroups/MyConfigurable.java
package krasa.editorGroups;

import com.intellij.openapi.options.Configurable;
import krasa.editorGroups.gui.SettingsForm;
import krasa.editorGroups.settings.EditorGroupsSettings;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class MyConfigurable implements Configurable {
  private SettingsForm form;


  @Nls
  @Override
  public String getDisplayName() {
    return "Editor Groups";
  }


  @Override
  @Nullable
  @NonNls
  public String getHelpTopic() {
    return null;
  }

  @Override
  public JComponent createComponent() {
    if (form == null) {
      form = new SettingsForm();
    }
    return form.getRoot();
  }


  @Override
  public boolean isModified() {
    return form != null && form.isSettingsModified(EditorGroupsSettings.getInstance());
  }

  @Override
  public void apply() {
    if (form != null) {
      form.apply();
    }
  }


  @Override
  public void reset() {
    if (form != null) {
      form.importFrom(EditorGroupsSettings.getInstance());
    }
  }

  @Override
  public void disposeUIResources() {
    form = null;
  }


}

File: src/main/java/krasa/editorGroups/MyConfigurable.kt
package krasa.editorGroups

import com.intellij.openapi.options.Configurable
import krasa.editorGroups.gui.SettingsForm
import krasa.editorGroups.messages.EditorGroupsBundle.message
import krasa.editorGroups.settings.EditorGroupsSettings.Companion.instance
import org.jetbrains.annotations.Nls
import org.jetbrains.annotations.NonNls
import javax.swing.JComponent

class MyConfigurable : Configurable {
  private var form: SettingsForm? = null

  override fun getDisplayName(): @Nls String = message("krasa.editorGroups.EditorGroupsSettings")

  override fun getHelpTopic(): @NonNls String? = null

  override fun createComponent(): JComponent? {
    if (form == null) form = SettingsForm()
    return form!!.root
  }

  override fun isModified(): Boolean = form != null && form!!.isSettingsModified(instance)

  override fun apply() {
    form?.apply()
  }

  override fun reset() {
    form?.importFrom(instance)
  }

  override fun disposeUIResources() {
    form = null
  }
}
